services/controllers/models.go

```diff
@@ -64,3 +76,38 @@ type (
 		Info              *watchmarket.Info `json:"info,omitempty"`
 	}
 )
+
+func GetCoinId(rawCoinId string) (uint, error) {
+	coinId, err := strconv.Atoi(rawCoinId)
+	if err != nil {
+		return 0, err
+	}
+	if _, ok := coin.Coins[uint(coinId)]; !ok {
+		return 0, errors.New(watchmarket.ErrBadRequest)
```

Comment:
```suggestion
		return 0, errors.New("invalid coin Id: " + coinId)
```

This is generic function, we should keep error separate from http error messages